embassy-nrf/src/twim.rs

```diff
@@ -703,14 +674,27 @@ pub enum Error {
     RxBufferTooLong,
     TxBufferZeroLength,
     RxBufferZeroLength,
-    Transmit,
-    Receive,
     DMABufferNotInDataMemory,
     AddressNack,
     DataNack,
     Overrun,
 }
 
+impl eh1::Error for Error {
+    fn kind(&self) -> eh1::ErrorKind {
+        match *self {
+            Self::TxBufferTooLong => eh1::ErrorKind::Other,
+            Self::RxBufferTooLong => eh1::ErrorKind::Other,
+            Self::TxBufferZeroLength => eh1::ErrorKind::Other,
+            Self::RxBufferZeroLength => eh1::ErrorKind::Other,
+            Self::DMABufferNotInDataMemory => eh1::ErrorKind::Other,
+            Self::AddressNack => eh1::ErrorKind::NoAcknowledge(eh1::NoAcknowledgeSource::Address),
+            Self::DataNack => eh1::ErrorKind::NoAcknowledge(eh1::NoAcknowledgeSource::Data),
+            Self::Overrun => eh1::ErrorKind::Overrun,
+        }
+    }
+}
+
```

Comment:
Such a pity to lose this fidelity of errors.

Reply:
It's how the EH ErrorKind is defined. 

It makes sense though: These errors are very nrf-specific. Users of embassy-nrf can still check our `Error` enum,  Hal-independent users via the traits can only check the kind, but it will have no idea about the nrf-specific errors, so there's nothing it can do anyway.